src/redux/reducers/TodoReducer.ts

```diff
@@ -0,0 +1,77 @@
+import { TodoItemType } from '../../models';
+import { AddItemPayload } from '../actions/todoActions/AddItemAction';
+import { ItemDeletePayload } from '../actions/todoActions/deleteItemAction';
+import { TodoActionTypes } from '../actions/todoActions';
+import { DoneItemPayload } from '../actions/todoActions/doneItemAction';
+import { SignInPayload } from '../actions/todoActions/signInAction';
+
+export type TodoReducerState = {
+  todoItems: TodoItemType[];
+  userToken: SignInPayload['userToken'] | null;
+  user: SignInPayload['user'] | null;
+};
+
+const initialState: TodoReducerState = {
+  todoItems: [],
+  userToken: null,
+  user: null,
+};
+
+export const todoReducer = (
+  state = initialState,
+  action: any,
```

Comment:
any not necessary

Reply:
if I remove this type I get an error:  Parameter 'action' implicitly has an 'any' type.